src/librustc_middle/ty/instance.rs

```diff
@@ -486,6 +481,42 @@ impl<'tcx> Instance<'tcx> {
             | InstanceDef::VtableShim(..) => Some(self.substs),
         }
     }
+
+    /// Returns a new `Instance` where generic parameters in `instance.substs` are replaced by
+    /// identify parameters if they are determined to be unused in `instance.def`.
+    pub fn polymorphize(self, tcx: TyCtxt<'tcx>) -> Self {
+        debug!("polymorphize: running polymorphization analysis");
+        if !tcx.sess.opts.debugging_opts.polymorphize {
+            return self;
+        }
+
+        if let InstanceDef::Item(..) = self.def {
+            let results = tcx.unused_generic_params(self.def_id());
```

Comment:
Hmm, why call `self.def_id()` when you can get the `DefId` from the `InstanceDef::Item(_)`?

Reply:
Fixed!